drivers/input/input_cap12xx.c

```diff
@@ -0,0 +1,230 @@
+
+	return 0;
+}
+
+static void cap12xx_work_handler(struct k_work *work)
+{
+	struct cap12xx_data *data = CONTAINER_OF(work, struct cap12xx_data, work);
+
+	cap12xx_process(data->dev);
+}
+
+static void cap12xx_timer_handler(struct k_timer *poll_timer)
+{
+	struct cap12xx_data *data = CONTAINER_OF(poll_timer, struct cap12xx_data, poll_timer);
+
+	k_work_submit(&data->work);
+}
+
+static void cap12xx_isr_handler(const struct device *dev, struct gpio_callback *cb, uint32_t pins)
+{
+	struct cap12xx_data *data = CONTAINER_OF(cb, struct cap12xx_data, int_gpio_cb);
+
+	k_work_submit(&data->work);
+}
+
+static int cap12xx_init(const struct device *dev)
+{
+	const struct cap12xx_config *config = dev->config;
+	struct cap12xx_data *data = dev->data;
+	int r;
+
+	if (!device_is_ready(config->i2c.bus)) {
+		LOG_ERR("I2C controller device not ready");
+		return -ENODEV;
+	}
+
+	data->dev = dev;
+
+	k_work_init(&data->work, cap12xx_work_handler);
+
+	if (config->int_gpio == NULL) {
+		LOG_DBG("cap12xx driver in polling mode");
+		k_timer_init(&data->poll_timer, cap12xx_timer_handler, NULL);
+		r = cap12xx_enable_interrupt(&config->i2c, true);
+		if (r < 0) {
+			LOG_ERR("Could not configure interrupt");
+			return r;
+		}
+		k_timer_start(&data->poll_timer, K_MSEC(config->poll_interval_ms),
+			      K_MSEC(config->poll_interval_ms));
+	} else {
+		LOG_DBG("cap12xx driver in interrupt mode");
+		if (!gpio_is_ready_dt(config->int_gpio)) {
+			LOG_ERR("Interrupt GPIO controller device not ready (missing device tree "
+				"node?)");
+			return -ENODEV;
+		}
+
+		r = gpio_pin_configure_dt(config->int_gpio, GPIO_INPUT);
+		if (r < 0) {
+			LOG_ERR("Could not configure interrupt GPIO pin");
+			return r;
+		}
+
+		r = gpio_pin_interrupt_configure_dt(config->int_gpio, GPIO_INT_EDGE_TO_ACTIVE);
+		if (r < 0) {
+			LOG_ERR("Could not configure interrupt GPIO interrupt");
+			return r;
+		}
+
+		gpio_init_callback(&data->int_gpio_cb, cap12xx_isr_handler,
+				   BIT(config->int_gpio->pin));
+
+		r = gpio_add_callback_dt(config->int_gpio, &data->int_gpio_cb);
+		if (r < 0) {
+			LOG_ERR("Could not set gpio callback");
+			return r;
+		}
+
+		r = cap12xx_clear_interrupt(&config->i2c);
+		if (r < 0) {
+			LOG_ERR("Could not clear interrupt");
+			return r;
+		}
+		r = cap12xx_enable_interrupt(&config->i2c, true);
+		if (r < 0) {
+			LOG_ERR("Could not configure interrupt");
+			return r;
+		}
+		if (config->repeat) {
+			r = i2c_reg_write_byte_dt(&config->i2c, REG_REPEAT_ENABLE, REPEAT_ENABLE);
+			if (r < 0) {
+				LOG_ERR("Could not disable repeated interrupts");
+				return r;
+			}
+			LOG_DBG("cap12xx enabled repeated interrupts");
+		} else {
+			r = i2c_reg_write_byte_dt(&config->i2c, REG_REPEAT_ENABLE, REPEAT_DISABLE);
+			if (r < 0) {
+				LOG_ERR("Could not enable repeated interrupts");
+				return r;
+			}
+			LOG_DBG("cap12xx disabled repeated interrupts");
+		}
+	}
+	LOG_DBG("cap12xx: %d channels configured", config->input_channels);
```

Comment:
no need for the cap12xx prefix here afaict

```suggestion
	LOG_DBG("%d channels configured", config->input_channels);
```